src/main/java/com/hms_networks/americas/sc/extensions/config/ConfigFileVersion.java

```diff
@@ -0,0 +1,201 @@
+      throw new IllegalArgumentException("Invalid version string");
+    }
+
+    try {
+      majorVersion = Integer.parseInt(matcher.group(MAJOR_VERSION_GROUP));
+      minorVersion = Integer.parseInt(matcher.group(MINOR_VERSION_GROUP));
+      patchVersion = Integer.parseInt(matcher.group(PATCH_VERSION_GROUP));
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Invalid version string");
+    }
+  }
+
+  /**
+   * Get the version string
+   *
+   * @return the version string
+   * @since 1.15.4
+   */
+  public String getVersionString() {
+    return versionString;
+  }
+
+  /**
+   * Get the major version
+   *
+   * @return the major version
+   * @since 1.15.4
+   */
+  public int getMajorVersion() {
+    return majorVersion;
+  }
+
+  /**
+   * Get the minor version
+   *
+   * @return the minor version
+   * @since 1.15.4
+   */
+  public int getMinorVersion() {
+    return minorVersion;
+  }
+
+  /**
+   * Get the patch version
+   *
+   * @return the patch version
+   * @since 1.15.4
+   */
+  public int getPatchVersion() {
+    return patchVersion;
+  }
+
+  /**
+   * Compare this ConfigFileVersion to another Object
+   *
+   * @param o the Object to compare to
+   * @return a negative integer, zero, or a positive integer as the specified version is greater
+   *     than, equal to, or less than this version.
+   * @since 1.15.4
+   */
+  public int compareTo(Object o) {
+    final int lessThanResult = -1;
+
+    if (o instanceof ConfigFileVersion) {
+      return compareTo((ConfigFileVersion) o);
+    } else {
+      return lessThanResult;
+    }
+  }
+
+  /**
+   * Compare this ConfigFileVersion to another ConfigFileVersion
+   *
+   * @param compConfigFileVersion the ConfigFileVersion to compare to
+   * @return a negative integer, zero, or a positive integer as the specified version is greater
+   *     than, equal to, or less than this version.
+   * @since 1.15.4
+   */
+  public int compareTo(ConfigFileVersion compConfigFileVersion) {
+    final int lessThanResult = -1;
+    final int greaterThanResult = 1;
+    final int equalResult = 0;
+
+    if (majorVersion > compConfigFileVersion.getMajorVersion()) {
+      return greaterThanResult;
+    } else if (majorVersion < compConfigFileVersion.getMajorVersion()) {
+      return lessThanResult;
+    } else {
+      if (minorVersion > compConfigFileVersion.getMinorVersion()) {
+        return greaterThanResult;
+      } else if (minorVersion < compConfigFileVersion.getMinorVersion()) {
+        return lessThanResult;
+      } else {
+        if (patchVersion > compConfigFileVersion.getPatchVersion()) {
+          return greaterThanResult;
+        } else if (patchVersion < compConfigFileVersion.getPatchVersion()) {
+          return lessThanResult;
+        } else {
+          return equalResult;
+        }
+      }
+    }
+  }
+
+  /**
+   * Get the string representation of this ConfigFileVersion.

```

Comment:
Class references could be `{@link ConfigFileVersion}` (there are a number of these throughout the file).

```suggestion
   * Get the string representation of this {@link ConfigFileVersion}.
```

Reply:
changed

Reply:
There are similar instances in `ConfigFileUpdateStep` and `ConfigFileUpdater`. My bad -- I should've documented those as well.

Reply:
added